core/src/word_lock.rs

```diff
@@ -137,6 +172,11 @@ impl WordLock {
             state = with_thread_data(|thread_data| {
                 // The pthread implementation is still unsafe, so we need to surround `prepare_park`
                 // with `unsafe {}`.
+                //
+                // REVIEW: this shouldn't be conditionally `unsafe` depending on
+                // the platform, it should have a uniform interface for all
+                // platforms with documented guarantees about how to make it
+                // safe on all platforms.
```

Comment:
I'm thinking about fixing both this and the fact that all `ThreadParker`s have duplicated documentation by creating a `ThreadParker` trait that all of them can implement. Then the documentation can be on the trait.

Reply:
That'd be great!